pymc_marketing/mmm/base.py

```diff
@@ -1274,6 +1274,124 @@ def plot_channel_contribution_share_hdi(
     def graphviz(self, **kwargs):
         return pm.model_to_graphviz(self.model, **kwargs)
 
+    def plot_waterfall_components_decomposition(
+        self, original_scale: bool = True, figsize: Tuple = (14, 7), **kwargs
+    ):
+        """
+        This function creates a waterfall plot. The plot shows the decomposition of the target into its components.
+
+        Parameters
+        ----------
+        original_scale : bool, optional
+            If True, the contributions are plotted in the original scale of the target.
+        figsize : Tuple, optional
+            The size of the figure. The default is (14, 7).
+        **kwargs
+            Additional keyword arguments to pass to the matplotlib `subplots` function.
+
+        Returns
+        -------
+        fig : matplotlib.figure.Figure
+            The matplotlib figure object.
+        ax : matplotlib.axes.Axes
+            The matplotlib axes object.
+        """
+
+        # Sort the dataframe in ascending order of contribution for the waterfall plot
+        dataframe = self.compute_mean_contributions_over_time(
+            original_scale=original_scale
+        )
+
+        dataframe["seasonal"] = (
+            dataframe["sin_order_1"]
+            + dataframe["sin_order_2"]
+            + dataframe["cos_order_1"]
+            + dataframe["cos_order_2"]
+        )
+        dataframe.drop(
+            ["sin_order_1", "sin_order_2", "cos_order_1", "cos_order_2"],
+            axis=1,
+            inplace=True,
+        )
```

Comment:
We should have a function that extract the seasonal components by looping over "cos_order_*" and "sin_order_*" over all components defined by `yearnly_seasonality` (https://github.com/pymc-labs/pymc-marketing/blob/main/pymc_marketing/mmm/delayed_saturated_mmm.py#L54)

Reply:
Yes, sounds good. Should we created in this same PR?

Reply:
I did the following and looks enough:

```python
        if getattr(self, "yearly_seasonality", None):
            contributions_fourier_over_time = pd.DataFrame(
                az.extract(
                    self.fit_result,
                    var_names=["fourier_contributions"],
                    combined=True,
                )
                .mean("sample")
                .to_dataframe()
                .squeeze()
                .unstack()
                .sum(axis=1),
                columns=["yearly_seasonality"],
            )
```

Reply:
yeah, you can do it on this PR